app/queries/find_exact_match.rb

```diff
@@ -0,0 +1,39 @@
+class FindExactMatch
+  def initialize(query_field: :tags)
+    raise ArgumentError.new("query_field must be `:tags` or `:links`") unless %i{tags links}.include?(query_field)
+
+    @query_field = query_field.to_sym
+  end
+
+
+  def call(query_hash, document_type)
```

Comment:
I'd make query_hash a keyword arg for greater clarity at call sites. But not document type, that's probably fine as it is.